src/llms-txt/infrastructure/content/manual-post-collection.php

```diff
@@ -0,0 +1,162 @@
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * The indexable helper.
+	 *
+	 * @var Indexable_Helper
+	 */
+	private $indexable_helper;
+
+	/**
+	 * The indexable repository.
+	 *
+	 * @var Indexable_Repository
+	 */
+	private $indexable_repository;
+
+	/**
+	 * The meta surface.
+	 *
+	 * @var Meta_Surface
+	 */
+	private $meta;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Options_Helper       $options_helper       The options helper.
+	 * @param Indexable_Helper     $indexable_helper     The indexable helper.
+	 * @param Indexable_Repository $indexable_repository The indexable repository.
+	 * @param Meta_Surface         $meta                 The meta surface.
+	 */
+	public function __construct(
+		Options_Helper $options_helper,
+		Indexable_Helper $indexable_helper,
+		Indexable_Repository $indexable_repository,
+		Meta_Surface $meta
+	) {
+		$this->options_helper       = $options_helper;
+		$this->indexable_helper     = $indexable_helper;
+		$this->indexable_repository = $indexable_repository;
+		$this->meta                 = $meta;
+	}
+
+	/**
+	 * The post method to get all relevant content type entries
+	 *
+	 * @param string $post_type The post type.
+	 * @param int    $limit     The maximum number of posts to return.
+	 *
+	 * @return array<int, array<Content_Type_Entry>> The posts that are relevant for the LLMs.txt.
+	 */
+	public function get_posts( string $post_type, int $limit ): array {
+		$posts = [];
+		$pages = [
+			'about_us_page',
+			'contact_page',
+			'terms_page',
+			'privacy_policy_page',
+			'shop_page',
+		];
+
+		foreach ( $pages as $page ) {
+			if ( ! empty( $this->options_helper->get( $page ) ) ) {
+				$page_id = $this->options_helper->get( $page );
+				if ( $this->indexable_helper->should_index_indexables() ) {
+					$post = $this->get_content_type_entry_for_indexable( $page_id );
+				}
+				else {
+					$post = $this->get_content_type_entry_wp_query( $page_id );
+				}
+				if ( $post !== null ) {
+					$posts[] = $post;
+				}
+			}
+		}
+
+		if ( ! empty( $this->options_helper->get( 'other_included_pages' ) ) ) {
+			foreach ( $this->options_helper->get( 'other_included_pages' ) as $page_id ) {
+				if ( $this->indexable_helper->should_index_indexables() ) {
+					$post = $this->get_content_type_entry_for_indexable( $page_id );
+				}
+				else {
+					$post = $this->get_content_type_entry_wp_query( $page_id );
+				}
+				if ( $post !== null ) {
+					$posts[] = $post;
+				}
+			}
+		}
+
+		return \array_slice( $posts, 0, $limit );
+	}
+
+	/**
+	 * Gets the content entries with WP query.
+	 *
+	 * @param int $page_id The id of the page.
+	 *
+	 * @return Content_Type_Entry The content type entry.
+	 */
+	public function get_content_type_entry_wp_query( int $page_id ): ?Content_Type_Entry {
+		$page = \get_post( $page_id );
+
+		if ( $page !== null && $page->post_password === '' && $page->post_status === 'publish' ) {
+
+			return new Content_Type_Entry(
```

Comment:
Same with the previous remark.  Maybe it's time we consider making an adapter that will create a content type entry out of a `WP_Post` object?